### ISSUE TITLE
Compatibility with mtl-2.3

### ISSUE THREAD

**Issue by sjakobi** [haskell-beam/beam#635]
beam-core will apparently need code changes:

```
Building library for beam-core-0.9.2.1..
[ 1 of 32] Compiling Database.Beam.Backend.Internal.Compat ( Database/Beam/Backend/Internal/Compat.hs, dist/build/Database/Beam/Backend/Internal/Compat.o, dist/build/Database/Beam/Backend/Internal/Compat.dyn_o )
[ 2 of 32] Compiling Database.Beam.Backend.SQL.Types ( Database/Beam/Backend/SQL/Types.hs, dist/build/Database/Beam/Backend/SQL/Types.o, dist/build/Database/Beam/Backend/SQL/Types.dyn_o )
[ 3 of 32] Compiling Database.Beam.Backend.Types ( Database/Beam/Backend/Types.hs, dist/build/Database/Beam/Backend/Types.o, dist/build/Database/Beam/Backend/Types.dyn_o )
[ 4 of 32] Compiling Database.Beam.Backend.SQL.Row ( Database/Beam/Backend/SQL/Row.hs, dist/build/Database/Beam/Backend/SQL/Row.o, dist/build/Database/Beam/Backend/SQL/Row.dyn_o )

Database/Beam/Backend/SQL/Row.hs:195:7: error:
    • Variable not in scope:
        replicateM_
          :: Int -> FromBackendRowM be0 () -> FromBackendRowM be b0
    • Perhaps you meant one of these:
        ‘Vector.replicateM’ (imported from Data.Vector.Sized),
        ‘Vector.replicateM'’ (imported from Data.Vector.Sized),
        ‘replicate’ (imported from Prelude)
    |
195 |       replicateM_ (valuesNeeded (Proxy @be) (Proxy @(Maybe x)))
    |       ^^^^^^^^^^^
```

**Comment by kmicklas** [comment]
That specific failure is indeed something we'll need to fix, but after that I hit errors relating to https://github.com/haskell/mtl/issues/117 which was an unintentional breakage in mtl 2.3 (which has now been deprecated on Hackage). After they release a new version without known issues I'll update Beam.

**Comment by kmicklas** [comment]
Fixed by #681.
